### PR TITLE
CI fix typing with latest ipython

### LINKED ISSUE #50543: CI latest ipython broke typing
```

mypy.....................................................................Failed
- hook id: mypy
- duration: 180.87s
- exit code: 1

pandas/io/formats/printing.py:264: error: Incompatible types in assignment (expression has type "str", base class "BaseFormatter" defined the type as "ObjectName")  [assignment]
pandas/io/formats/printing.py:265: error: Incompatible types in assignment (expression has type "Tuple[Type[Dict[Any, Any]]]", base class "BaseFormatter" defined the type as "Type[str]")  [assignment]
pandas/tests/test_register_accessor.py:16: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
pandas/tests/io/xml/test_xml.py:1118: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
pandas/tests/io/xml/test_xml.py:1297: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
pandas/tests/io/xml/test_to_xml.py:995: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 2 errors in 1 file (checked 1396 source files)
```

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #50543 

ref

https://github.com/ipython/ipython/blob/01bc9d96e0dd50fad02d7b1e3d2e58113c617b6b/IPython/core/formatters.py#L777-L792

### CODE REVIEW

File: pandas/io/formats/printing.py [hunk 1]
```diff
@@ -259,9 +259,10 @@ def enable_data_resource_formatter(enable: bool) -> None:
         if mimetype not in formatters:
             # define tableschema formatter
             from IPython.core.formatters import BaseFormatter
+            from traitlets import ObjectName
```
Comment (mroeschke) [inline pandas/io/formats/printing.py:262]: We don't declare traitlets as a dependency as it comes from IPython?

Comment (MarcoGorelli) [reply pandas/io/formats/printing.py:262]: that's right!

**Review by datapythonista — APPROVED** [review]
Not so familiar with traitlets, but seems like the CI is happy with this, lgtm

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @MarcoGorelli

**Pull request merged**